src/unix/linux_like/android/b32/arm.rs

```diff
@@ -1,5 +1,128 @@
 pub type c_char = u8;
 pub type wchar_t = u32;
+pub type greg_t = i32;
+
+s! {
+    pub struct sigcontext {
+        pub trap_no: ::c_ulong,
+        pub error_code: ::c_ulong,
+        pub oldmask: ::c_ulong,
+        pub arm_r0: ::c_ulong,
+        pub arm_r1: ::c_ulong,
+        pub arm_r2: ::c_ulong,
+        pub arm_r3: ::c_ulong,
+        pub arm_r4: ::c_ulong,
+        pub arm_r5: ::c_ulong,
+        pub arm_r6: ::c_ulong,
+        pub arm_r7: ::c_ulong,
+        pub arm_r8: ::c_ulong,
+        pub arm_r9: ::c_ulong,
+        pub arm_r10: ::c_ulong,
+        pub arm_fp: ::c_ulong,
+        pub arm_ip: ::c_ulong,
+        pub arm_sp: ::c_ulong,
+        pub arm_lr: ::c_ulong,
+        pub arm_pc: ::c_ulong,
+        pub arm_cpsr: ::c_ulong,
+        pub fault_address: ::c_ulong,
+    }
+}
+
+s_no_extra_traits! {
+    pub struct ucontext_t {
+        pub uc_flags: ::c_ulong,
+        pub uc_link: *mut ucontext_t,
+        pub uc_stack: ::stack_t,
+        // NOTE This field's type is actually mcontext_t,
+        //      which is a synonym for sigcontext.
+        //      But to avoid the error reported by style checker,
+        //      it must be sigcontext.
+        pub uc_mcontext: sigcontext,
```

Comment:
What error is this? Using `mcontext_t` should work just fine?

Reply:
At first I wrote [like this](https://github.com/rust-lang/libc/pull/1622/commits/546a690047f01e85dd50f5b0967fa03be5294a3b#diff-8c71e51ad9afb09c731c4ec6ac99ed44R31-R41):

```
pub type mcontext_t = sigcontext;

cfg_if! {
    if #[cfg(libc_union)] {
        s_no_extra_traits! {

            pub struct ucontext_t {
...
                pub uc_mcontext: mcontext_t,
...
```

But I got this error:

```
src/unix/linux_like/android/b32/arm.rs:31 - typedef found after struct when it belongs before
```

from https://dev.azure.com/rust-lang2/libc/_build/results?buildId=949&view=logs&j=c5397555-4577-58d8-d518-821f0664b4e0&t=01cef1e2-802e-5c67-cfe4-12bd866a84b0&l=16

I guess this is because the macros are expanded and evaluated before `pub type mcontext_t = sigcontext;` is evaluated.

So I tried to move `pub type mcontext_t = sigcontext;` inside `s_no_extra_traits!` block.
But that wasn't compilable because `s_no_extra_traits!` can't match `pub type ...`.
Maybe I should fix `s_no_extra_traits!`, but I'm not sure about Rust's macros.

Reply:
> But I got this error:

That just means that you need to group the `type = ...` with all other `type = ...` at the beginning of the file.

Reply:
That error could be arguably much better, sorry about that 😆